### PR TITLE
feat(component): add carousel component

### PR DESCRIPTION
**Pull request by tleperou**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Follow up of @gederer #253 .

Following discussion with Greg; here's an implementation using child components and context.

It support any type of content ✌️ 

## Includes
* aria compliant
* keyboard control
* touch compatible
* mouse friends
* uses browsers' scroll API
* navigation and next/previous buttons
* custom initial active item
* Axe tests passed

**Note**: the identification of the active item while scrolling is made by using the intersection observer API.

Usage

```tsx
<Carousel.Root startAt={1} loop={false}>
    <Carousel.ButtonPrevious>prev</Carousel.ButtonPrevious>
    <Carousel.Items>
      {items.value.map((_, i) => (
        <Carousel.Item key={useId()} index={i} label={`Accessible name of ${i}`}>
          // .. content here
        </Carousel.Item>
      ))}
    </Carousel.Items>
    <Carousel.ButtonNext>next</Carousel.ButtonNext>
  <Carousel.Controls>
    {items.value.map((_, i) => (
      <Carousel.Control key={useId()} index={i}>{i + 1}</Carousel.Control>
    ))}
  </Carousel.Controls>
</Carousel.Root>
```

# Use cases and why

Anywhere a Carousel is needed.

# Screenshots/Demo

image

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by gioboa** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you all sign our [Contributor License Agreement](https://github.com/qwikifiers/qwik-ui/blob/main/CLA.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> I have read the CLA Document and I hereby sign the CLA
> 
> **1** out of **2** committers have signed the CLA.✅ @tleperou❌ @gedererYou can retrigger this bot by commenting **recheck** in this Pull Request

@gederer can you sign the CLA pls?

File: packages/headless/src/components/carousel/use-ordinal.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { noSerialize } from '@builder.io/qwik';
+
+export const useOrdinal = () => {
```
Comment (gioboa) [inline packages/headless/src/components/carousel/use-ordinal.ts:3]: can you move this methods into the shared/utils folder please?

**Comment by gederer** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by gioboa** [comment]
@all-contributors please add @gederer for code

**Pull request merged**